mathematica/integrator_plots.cpp

```diff
@@ -110,40 +114,46 @@ void GenerateSimpleHarmonicMotionWorkErrorGraphs() {
     std::vector<Energy> e_errors;
     std::vector<double> evaluations;
     for (int i = 0; i < 500; ++i, parameters.Δt /= step_reduction) {
-      LOG_IF(INFO, (i + 1) % 10 == 0) << parameters.Δt;
+      int const number_of_evaluations =
+          method.stages * std::floor(parameters.tmax / parameters.Δt);
+      LOG_IF(INFO, (i + 1) % 50 == 0) << number_of_evaluations;
       method.integrator->SolveTrivialKineticEnergyIncrement<Length>(
           &ComputeHarmonicOscillatorAcceleration,
           parameters,
           &solution);
-      q_errors.emplace_back(
-          AbsoluteError(q_amplitude * Cos(ω * solution[0].time.value),
-                        solution[0].positions[0].value));
-      v_errors.emplace_back(
-          AbsoluteError(-v_amplitude * Sin(ω * solution[0].time.value),
-                        solution[0].momenta[0].value));
-      e_errors.emplace_back(
-          AbsoluteError(
-              0.5 * Joule,
-              (m * Pow<2>(solution[0].momenta[0].value) +
-               k * Pow<2>(solution[0].positions[0].value)) / 2));
-      evaluations.emplace_back(
-          method.stages * parameters.tmax / parameters.Δt);
+      std::vector<Length> q_error;
+      std::vector<Speed> v_error;
+      std::vector<Energy> e_error;
+      for (auto const& system_state : solution) {
+        q_error.emplace_back(
+            AbsoluteError(q_amplitude * Cos(ω * system_state.time.value),
+                          system_state.positions[0].value));
+        v_error.emplace_back(
+            AbsoluteError(-v_amplitude * Sin(ω * system_state.time.value),
+                          system_state.momenta[0].value));
+        e_error.emplace_back(
+            AbsoluteError(
+                0.5 * Joule,
+                (m * Pow<2>(system_state.momenta[0].value) +
+                 k * Pow<2>(system_state.positions[0].value)) / 2));
+      }
+      // We plot the maximum error, i.e., the L∞ norm of the error.
+      // Blanes and Moan (2002), or Blanes, Casas and Ros (2001) tend to use
+      // the average error (the normalized L¹ norm) instead.
+      q_errors.emplace_back(*std::max_element(q_error.begin(), q_error.end()));
+      v_errors.emplace_back(*std::max_element(v_error.begin(), v_error.end()));
+      e_errors.emplace_back(*std::max_element(e_error.begin(), e_error.end()));
+      evaluations.emplace_back(number_of_evaluations);
     }
     q_plots.emplace_back(PlottableDataset(evaluations, q_errors));
     v_plots.emplace_back(PlottableDataset(evaluations, v_errors));
     e_plots.emplace_back(PlottableDataset(evaluations, e_errors));
     names.emplace_back(Escape(method.name));
   }
-  file << Assign("qErrorPlots", q_plots);
-  file << Assign("vErrorPlots", v_plots);
-  file << Assign("eErrorPlots", e_plots);
+  file << Assign("qErrorData", q_plots);
```

Comment:
I'd change the name of these variables from _plots to _data for consistency with the generated Mathematica.